### DISCUSSION TITLE
Has any team applied remote IDE for the Android project?

### DISCUSSION THREAD

**Discussion by roscrazy** [MobileNativeFoundation/discussions#57] [General]
When mobile projects get large with a few million LOC, thousands of modules, and hundreds of libraries, the IDE experience worsens. As title, how do you improve the IDE experience for your team?

From my side, besides changing some config to optimize the performance to Android Studio (or IntelliJ IDE). Recently, I would like to discover the positives of the remote IDE. There are a few options I planned to discover:
- [Jetbrains projector](https://jetbrains.github.io/projector-client/mkdocs/latest/) -> Still in preview stage
- [VS Code Remote Development ](https://code.visualstudio.com/docs/remote/remote-overview) -> I hear the user experience with Visual Studio for Android project is not so good
- [Shared Indexes Plugin ](https://blog.jetbrains.com/idea/2020/07/shared-indexes-plugin-unveiled/) -> It's not a fully remote IDE solution, but remote indexing would help on the IDE experience.

If any team already applies remote IDE to your Android project, I strongly hope you could share a bit of your experience and the result.

**Comment by arturdryomov** [comment]
Not exactly the IDE experience but...

At Juno (RIP) we’ve created [Mainframer](https://github.com/buildfoundation/mainframer) and used it with great success over a couple of years. The idea is simple — have a remote machine (a _mainframe_) with powerful hardware and use it across the team. Even with the network overhead it performed better than the local execution. It was especially useful in early Kotlin days when there was no (zero) incremental compilation. ATM neither Mainframer nor [the IJ plugin](https://github.com/elpassion/mainframer-intellij-plugin) are actively supported though. It will not help with the IDE side of things but will offload the build itself which will give the hardware a breathing room. Nowadays the future is with the build system-level remote execution ([see Bazel](https://docs.bazel.build/versions/master/remote-execution.html)).

Also there is [an SSH VS Code plugin](https://marketplace.visualstudio.com/items?itemName=ms-vscode-remote.remote-ssh) but I cannot imagine anyone brave enough to use VS Code for Android development.

**Reply by roscrazy** [reply]
Hi @arturdryomov , thanks a lot for your reply. 
Yeah. I totally agree with everything you said. We also use Mainframer and on the transition to migrate to use Bazel too (almost there). But they both mostly only help for the build time. 
I would like to improve the IDE experiecne. It's a bit lagging/slow when you modify any file in a module at the top of the dependencies graph and the sync/indexing time is not so good.
Lastly your [SSH VS Code plugin](https://marketplace.visualstudio.com/items?itemName=ms-vscode-remote.remote-ssh) actually is the [VS Code Remote Development ](https://code.visualstudio.com/docs/remote/remote-overview) that I mentioned. And I also hear it's not so good for Android dev
